services/src/Microsoft.Azure.IIoT.Services.OpcUa.Publisher/src/Startup.cs

```diff
@@ -66,7 +66,8 @@ public class Startup {
         /// </summary>
         /// <param name="env"></param>
         /// <param name="configuration"></param>
-        public Startup(IWebHostEnvironment env, IConfiguration configuration) :
+        /// <param name="logger"></param>
+        public Startup(IWebHostEnvironment env, IConfiguration configuration, ILogger logger) :
```

Comment:
Can you please validate that `Microsoft.Azure.IIoT.Services.OpcUa.Publisher` project is able to start and run ? DI container builder is not configured yet at this point, so I am not sure that it will be able to resolve `ILogger` dependency.

I got this error, when I only added `ILogger` parameter for this constructor of `Startup`:

```
Exception thrown: 'System.InvalidOperationException' in Microsoft.Extensions.DependencyInjection.Abstractions.dll
An unhandled exception of type 'System.InvalidOperationException' occurred in Microsoft.Extensions.DependencyInjection.Abstractions.dll
Unable to resolve service for type 'Serilog.ILogger' while attempting to activate 'Microsoft.Azure.IIoT.Services.OpcUa.Publisher.Startup'.
```

Reply:
Hmm, the standalone publisher works, but you are right, the service doesn't and it doesn't look good: [Generic Host restricts Startup constructor injection](https://github.com/dotnet/aspnetcore/issues/9337#issuecomment-539859667). I will look into it.

Reply:
With current changes OPC Publisher microservice works fine.